### PR TITLE
Swipe To Delete Favorite Routes and Stops

### LINKED ISSUE #379: Swipe To Delete Favorites On My Buses
In PVTrAck 1.x, one could swipe left on their favorites on MyBuses and be greeted with a way to delete them, like so:
merp

PVTrAck 2.x doesn't have this yet, and would be a nice little addition.  Doing so will require small changes to the MyBuses view and component, and will familiarize you with some of Ionic's HTML directives, event handing in TypeScript, and our custom Favorites schemas.

### PR DESCRIPTION
**Pull request by AkivaGreen**
I want to resolve #379 with this pull request.

### CODE REVIEW

**Comment by AkivaGreen** [comment]
I'm looking for a way to refresh the interface after deleting to show that the route/stop has been deleted. Also, it does delete the route/stop, however, I don't know if it is doing it correctly.

**Comment by werebus** [comment]
Looks promising. If I understand the Ionic docs correctly, this isn't exactly a swipe-to-delete, but a swipe-to-reveal-a-delete-button?

@akaplo had a tool to record animated-GIF screencasts that was useful in these UI demonstration situations.

Also, we should look into the build failure. It's not caused by your code, as far as I can tell.

**Comment by akaplo** [comment]
Looks promising indeed! I don't have time today, but I will review in full this weekend.

**Comment by AkivaGreen** [comment]
The only thing I have left is getting it to remove the 'deleted' item from the list without having to do a browser refresh. Or should I do a browser refresh?

**Comment by akaplo** [comment]
@AkivaGreen, since MyBuses has  [instance-variable arrays](https://github.com/umts/pvta-multiplatform/blob/master/src/pages/my-buses/my-buses.component.ts#L22) for both favorite routes and favorite stops, removing the favorite in question from the proper array would solve that, as the view would auto-update.

You can see examples of how we did this in 1.x [here](https://github.com/umts/pvta-multiplatform/blob/ionic1/base/www/pages/my-buses/my-buses-controller.js#L79).

@werebus, CI failure is still because of #378.

File: src/pages/my-buses/my-buses.component.ts [hunk 1]
```diff
@@ -25,7 +27,8 @@ export class MyBusesComponent {
   trips;
   constructor(public navCtrl: NavController, private storage: Storage,
     private alertSvc: AlertService, private alertCtrl: AlertController,
-    private modalCtrl: ModalController, private tripSvc: FavoriteTripService) {
+    private modalCtrl: ModalController, private tripSvc: FavoriteTripService,
+    private stopSvc: FavoriteStopService, private routeSvc: FavoriteRouteService) {
```
Comment (akaplo) [inline src/pages/my-buses/my-buses.component.ts]: Your naming for the service instances could be a little confusing - for example, the app has a [RouteService](https://github.com/umts/pvta-multiplatform/blob/master/src/providers/route.service.ts) **and** [FavoriteRouteService](https://github.com/umts/pvta-multiplatform/blob/master/src/providers/favorite0route.service.ts).  

I'd alleviate this with names like `favRouteSvc` and `favStopSvc`.

**Review by akaplo — CHANGES_REQUESTED** [review]
I know you're still working on deleting the favorites, but I'd also love to see some unit tests for the new methods!

**Comment by akaplo** [comment]
@AkivaGreen what's the status here?  Let me know if you're ready for me to look it over again.

**Comment by mboneil10** [comment]
@akaplo, @AkivaGreen gave me this PR/issue. He's currently working on something over in round-three.

**Comment by mboneil10** [comment]
Okay, so from my understanding of this issue, all I have to do is use an instance variable array for favorite routes/stops and supply unit tests? @akaplo @AkivaGreen

**Comment by mboneil10** [comment]
I've made a [new PR for this issue](https://github.com/umts/pvta-multiplatform/pull/408), so I'll close this one. Good work @AkivaGreen!

**Pull request closed without merging**